documentation/sphinx/source/developer-guide.rst

```diff
@@ -955,3 +1067,5 @@ The trickiest errors are non-retryable errors. ``Transaction.on_error`` will ret
 If you see one of those errors, the best way of action is to fail the client.
 
 At a first glance this looks very similar to an ``commit_unknown_result``. However, these errors lack the one guarantee ``commit_unknown_result`` still gives to the user: if the commit has already been sent to the database, the transaction could get committed at a later point in time. This means that if you retry the transaction, your new transaction might race with the old transaction. While this technically doesn't violate any consistency guarantees, abandoning a transaction means that there are no causality guaranatees.
+
+.. [#conflicting_keys] In practice it probably committed successfully, but if you're running multiple resolvers then it's possible that a transaction can cause another transaction to abort without committing successfully.
```

Comment:
I'm a bit confused at the relation between this sentence and this section "Non-retryable errors". Do you mean error on `conflict_keys` is non-retryable error, because ...?

Reply:
It's a footnote

Reply:
Oh, I see. Would this sentence be clearer by specifying 'it'? say, 
In practice, the write transaction probably committed successfully. However, if you're running multiple resolvers then it's possible that another transaction can cause the write transaction to abort without committing successfully.

Reply:
I feel this is a subtle point, should be very helpful.